### PR TITLE
Added better error messages for vmware tests

### PR DESCRIPTION
**Pull request by Akm0d**
### What does this PR do?
Add more descriptive errors for vmware tests

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/53011

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/integration/cloud/clouds/test_vmware.py [hunk 1]
```diff
@@ -88,7 +88,11 @@ def test_instance(self):
         profile_config = cloud_config(profile)
         disk_datastore = profile_config['vmware-test']['devices']['disk']['Hard disk 2']['datastore']
 
+        for key, value in profile_config.get('vmware-test', {}).items():
+            assert value, "Instance config incomplete; missing '{}'".format(key)
```
Comment (waynew) [inline tests/integration/cloud/clouds/test_vmware.py]: Do you want to blow up on the first problem here, or would it be better to collect potential issues?

Comment (Akm0d) [reply tests/integration/cloud/clouds/test_vmware.py]: The errors I was getting before led me in endless circles, so yes, I want to blow up sooner

**Review by waynew — APPROVED** [review]

File: tests/integration/cloud/clouds/test_vmware.py [hunk 2]
```diff
@@ -88,7 +88,11 @@ def test_instance(self):
         profile_config = cloud_config(profile)
         disk_datastore = profile_config['vmware-test']['devices']['disk']['Hard disk 2']['datastore']
 
+        for key, value in profile_config.get('vmware-test', {}).items():
```
Comment (rares-pop) [inline tests/integration/cloud/clouds/test_vmware.py]: is fine for a test, but typically it's better to use six.iteritems() instead of items()

Comment (Akm0d) [reply tests/integration/cloud/clouds/test_vmware.py]: I'll make that change

**Review by rares-pop — APPROVED** [review]

**Comment by Akm0d** [comment]
re-run windows-2016

**Comment by Akm0d** [comment]
re-run windows

**Pull request merged**